src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1149,545 @@ struct Wallet
+    // - tools::wallet2::pool_payment_details
+    // - tools::wallet2::signed_tx
+    // - cryptonote::address_parse_info
+// TODO : wallet2::transfer_container
+    /**
+    * brief: getTransfers - get all transfers
+    * outparam: transfers -
+    */
+//    virtual void getTransfers(wallet2::transfer_container& transfers) const = 0;
+// TODO : wallet2::payment_details
+    /**
+    * brief: getPayments - get incoming transfers
+    * param: payment_id -
+    * param: payments -
+    * param: min_height -
+    * param: subaddr_account -
+    * param: subaddr_indices -
+    */
+//    virtual void getPayments(const std::string &payment_id, std::list<wallet2::payment_details> &payments, std::uint64_t min_height, const boost::optional<uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::payment_details
+    /**
+    * brief: getPayments - get incoming transfers
+    * param: payments -
+    * param: max_height -
+    */
+//    virtual void getPayments(std::list<std::pair<std::string, wallet2::payment_details>> &payments, std::uint64_t min_height, std::uint64_t max_height, const boost::optional<uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::confirmed_transfer_details
+    /**
+    * brief: getPaymentsOut - get outgoing transfers
+    * param: confirmed_payments -
+    */
+//    virtual void getPaymentsOut(std::list<std::pair<std::string, wallet2::confirmed_transfer_details>> &confirmed_payments, std::uint64_t min_height, std::uint64_t max_height, const boost::optional<std::uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::unconfirmed_transfer_details
+    /**
+    * brief: getUnconfirmedPaymentsOut - get unconfirmed transfers
+    * param: unconfirmed_payments -
+    */
+//    virtual void getUnconfirmedPaymentsOut(std::list<std::pair<std::string, wallet2::unconfirmed_transfer_details>> &unconfirmed_payments, const boost::optional<std::uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::pool_payment_details
+    /**
+    * brief: getUnconfirmedPayments - get pending transfers, currently in pool
+    * param: unconfirmed_payments -
+    */
+//    virtual void getUnconfirmedPayments(std::list<std::pair<std::string, wallet2::pool_payment_details>> &unconfirmed_payments, const boost::optional<std::uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+    /**
+    * brief: dumpMultisigTxToString - get the encrypted unsigned multisig transaction as hex string from a multisig pending transaction
+    * param: multisig_ptx - multisig pending transaction
+    * return: encrypted tx data as hex string if succeeded, else empty string
+    * note: sets status error on fail
+    */
+    virtual std::string dumpMultisigTxToStr(const PendingTransaction &multisig_ptx) const = 0;
```

Comment:
As a terminology question, at first sight I am not sure what to make of the term *dumping* that gets used here. Maybe it's established in the codebase, don't have an overview right now. I ask myself why not simply use *convert*, and use it for all possible cases where some info just changes form.

Reply:
It's not established afaik, just chose it to differentiate between the two different types of `wallet2::save_multisig_tx()`, one just returns the encrypted hex string (that's what I called "dump") and the other one actually saves the hex string. I'm fine with changing it to "convert".

Reply:
Correction: not established in the API, but in wallet2 we have `dump_tx_to_str` which is the equivalent function for the multisig `save_multisig_tx` which returns a string as mentioned above.
Still fine to go with _convert_ here instead of _dump_ imo.